osu.Game/Graphics/Carousel/Carousel.cs

```diff
@@ -514,8 +514,29 @@ public bool OnPressed(KeyBindingPressEvent<GlobalAction> e)
             }
 
             return false;
+
+            void traverseFromKey(TraversalOperation traversal)
+            {
+                switch (traversal.Type)
+                {
+                    case TraversalType.Keyboard:
+                        traverseKeyboardSelection(traversal.Direction);
+                        break;
+
+                    case TraversalType.Set:
+                        traverseSetSelection(traversal.Direction);
+                        break;
+
+                    default:
+                        throw new ArgumentOutOfRangeException();
+                }
+            }
         }
 
+        private enum TraversalType { Keyboard, Set }
```

Comment:
Shouldn't this be

```suggestion
        private enum TraversalType { Keyboard, Set, Group }
```

now, and shouldn't the code paths added in https://github.com/ppy/osu/pull/33668 use that new `Group` member?

Reply:
Yep, will get to that today 👍